rocketchat-core/src/main/java/com/rocketchat/core/factory/ChatRoomFactory.java

```diff
@@ -14,16 +18,21 @@
 // TODO: 29/7/17 might have to make separate arraylist for each type, seems little impossible, better to keep generic
 public class ChatRoomFactory {
 
-    private RocketChatClient client;
-    private List<ChatRoom> rooms;
+    private RocketChatClient api;
+    private ArrayList<ChatRoom> rooms;
```

Comment:
Always use the interface `List` as the declaration... This way we can change the internal implementation to `LinkedList` for example, and the user will still see a `List` and won't need to change anything.

Reply:
Will do it.